# <b> Results from 2023 Financial Reports (Q1, Q2, Q3, Q4)</b>

In [9]:
#testing QTR1 first

import pandas as pd
df = pd.read_csv("/Users/rravindra0463/Desktop/ai_proj/venv/new_output/New_Results_QTR1.csv")
df.describe()

filename  \
count                                                 388   
unique                                                388   
top     20230303_10-K_edgar_data_1403256_0001403256-23...   
freq                                                    1   

                                                   Item1A  \
count                                                 388   
unique                                                387   
top     . Risk Factors You should carefully consider t...   
freq                                                    2   

                                                 Sentence  \
count                                                 388   
unique                                                373   
top     In addition, our implementation of certain new...   
freq                                                    3   

                                              SentencesR3  \
count                                                 388   
unique                                                383   
top     A number of recent technologies have worked wi...   
freq                                                    3   

                                              SentencesR5  \
count                                                 388   
unique                                                386   
top     Summary of Risk Factors These statements refle...   
freq                                                    2   

               KeywordMentioned  
count                       388  
unique                      139  
top     artificial intelligence  
freq                         81

So about 388 files are ones which have AI related keywords in them.

In [10]:
# check how many files orginally present in the concerned folder,i.e QTR1 here
import os
folder_path = '/Users/rravindra0463/Desktop/ai_proj/venv/QTR1'

# list all files in the folder
files = [f for f in os.listdir(folder_path) if os.path.isfile(os.path.join(folder_path, f))]

# count the number of files
num_files = len(files)
print(f'Number of files in the folder QTR1: {num_files}')

# calculate metrics - using unique to ensure there are no duplicates 
unique_filenames = df['filename'].nunique()
percentage = (unique_filenames/ num_files) * 100

print(f'Number of relevant files extracted: {unique_filenames}')
print(f'Percentage of relevant files: {percentage:.2f}%')

Number of files in the folder QTR1: 7145
Number of relevant files extracted: 388
Percentage of relevant files: 5.43%


So out of 7k+ financial reports, about 390 are those whose risk factors are related to AI.

In [11]:
df.head()

filename  \
0  20230301_10-K_edgar_data_1831915_0000950170-23...   
1  20230221_10-K_edgar_data_1410384_0001410384-23...   
2  20230228_10-K_edgar_data_1315098_0001315098-23...   
3  20230113_10-K_edgar_data_1829966_0001683168-23...   
4  20230217_10-K_edgar_data_871763_0000950170-23-...   

                                              Item1A  \
0  . Risk Factors Our operations and financial re...   
1  . Risk Factors. Our business, prospects, finan...   
2  . Risk Factors RISK FACTORS A description of t...   
3  . Risk Factors. An investment in our securitie...   
4  . Ri sk Factors FORWARD-LOOKING STATEMENTS Thi...   

                                            Sentence  \
0  Furthermore, in Europe, there is a proposed re...   
1  Our principal risks include risks associated w...   
2  The Roblox Cloud may be relied upon in the fut...   
3  We have a new business model and are in the pr...   
4  We expect our use of data to increase, includi...   

                                         SentencesR3  \
0  For example, the European Union s General Data...   
1  Risk Factors. Our business, prospects, financi...   
2  In the past, it has been difficult to remove f...   
3  There can be no assurance that we will generat...   
4  This may worsen as clients increasingly take a...   

                                         SentencesR5  \
0  If we become subject to new data privacy and s...   
1  . Risk Factors. Our business, prospects, finan...   
2  When we develop new or enhanced features for o...   
3  As a result, we have generated limited revenue...   
4  We expect that we will continue to experience ...   

                                    KeywordMentioned  
0                            artificial intelligence  
1                 machine learning, machine learning  
2  generative ai, artificial intelligence, machin...  
3                              business intelligence  
4  artificial intelligence, machine learning, art...

In [12]:
df.tail()

filename  \
383  20230316_10-K_edgar_data_1807846_0000950170-23...   
384  20230223_10-K_edgar_data_1801169_0001801169-23...   
385  20230224_10-K_edgar_data_927628_0000927628-23-...   
386  20230331_10-K_edgar_data_1812360_0001213900-23...   
387  20230303_10-K_edgar_data_1403256_0001403256-23...   

                                                Item1A  \
383  . Risk Factors Risks Relating to Our Business ...   
384  . Risk Factors. In the course of conducting ou...   
385  . Risk Factors The following discussion sets f...   
386  . Risk Factors The following risks could mater...   
387  . Risk Factors In the course of conducting our...   

                                              Sentence  \
383  We may also face claims from third parties cla...   
384  We appraise and price the homes we buy and sel...   
385  The use of artificial intelligence, bots or ot...   
386  In addition, new and changed rules and regulat...   
387  These impacts may include venue trading requir...   

                                           SentencesR3  \
383  Such open-source software may include software...   
384  These differences may result in greater pricin...   
385  There has also been a significant proliferatio...   
386  Even if not subject to legal challenge, the pe...   
387  MiFID II and UK MiFID II have imposed signific...   

                                           SentencesR5  \
383  Some aspects of our business processes include...   
384  Furthermore, in order to grow our business, we...   
385  Cyber and information security risks for large...   
386  Despite our efforts to comply with applicable ...   
387  MiFID II, which is comprised of the Markets in...   

                                      KeywordMentioned  
383  artificial intelligence, machine learning, mac...  
384                                       data science  
385  artificial intelligence, artificial intelligen...  
386          artificial intelligence, machine learning  
387                                algorithmic trading

Randomly Checking the content extracted - sentence & the AI term used: 

In [17]:
content = df.loc[1,'Item1A']
print(content)

. Risk Factors. Our business, prospects, financial condition, operating results and the trading price of our common stock could be materially adversely affected by a variety of risks and uncertainties, including those described below, as well as other risks not currently known to us or that are currently considered immaterial. In assessing these risks, you should also refer to the other information contained in this Annual Report on Form -K, including our consolidated financial statements and related notes. Our principal risks include risks associated with: our ability to manage our rapid growth; the challenging macroeconomic environment, including its impact on our customers' purchasing decisions on products and services which are more discretionary in nature and the related demand for our solutions relative to our expectations; the length, cost and unpredictability of our sales cycle; the development of and changes to the market for our solutions compared to our expectations; our abi

In [22]:
Sentence = df.loc[1,'Sentence']
print(Sentence)

keyword = df.loc[1,"KeywordMentioned"]
print(keyword)

Our principal risks include risks associated with: our ability to manage our rapid growth; the challenging macroeconomic environment, including its impact on our customers' purchasing decisions on products and services which are more discretionary in nature and the related demand for our solutions relative to our expectations; the length, cost and unpredictability of our sales cycle; the development of and changes to the market for our solutions compared to our expectations; our ability to attract new customers and expand and renew existing customer relationships; focusing on the financial services industry and any geographies where we have customer concentration; managing challenges and costs associated with the implementation of a higher volume of or more complex configurations of our solutions; customer acceptance of and satisfaction with our existing and new solutions; the strength of our brand and reputation; intense competition in the markets we serve and challenges we face as we

In [23]:
content2 = df.loc[4,'Sentence']
print(content2)
keyword2 = df.loc[4,"KeywordMentioned"]
print(keyword)

We expect our use of data to increase, including through the use of analytics, artificial intelligence (AI) and machine learning (ML). For example, rapid changes and regulatory restrictions on the use of artificial intelligence, machine learning and robotics are having a significant impact on some of the industries we serve and could have significant and unforeseen consequences for the workforce services industry and for our business.
machine learning, machine learning


In [25]:
Sentence2 = df.loc[1,'Sentence']
print(Sentence)

Keyword2 = df.loc[1,'KeywordMentioned']
print(Keyword2)

Our principal risks include risks associated with: our ability to manage our rapid growth; the challenging macroeconomic environment, including its impact on our customers' purchasing decisions on products and services which are more discretionary in nature and the related demand for our solutions relative to our expectations; the length, cost and unpredictability of our sales cycle; the development of and changes to the market for our solutions compared to our expectations; our ability to attract new customers and expand and renew existing customer relationships; focusing on the financial services industry and any geographies where we have customer concentration; managing challenges and costs associated with the implementation of a higher volume of or more complex configurations of our solutions; customer acceptance of and satisfaction with our existing and new solutions; the strength of our brand and reputation; intense competition in the markets we serve and challenges we face as we

Hence, here AI Sentence extracted from risk factors is: 
"...including evolving regulation of AI, machine learning and the receipt, collection, storage,..." and keyword is "machine learning"

Let's see what unique keywords have been extracted to double-verify if they are AI related:

In [29]:
# Initialize an empty set to store unique keywords
unique_keywords = set()

# Iterate through each row in the 'KeywordMentioned' column
for keywords in df['KeywordMentioned'].dropna():
    # Split the keywords by comma and strip any extra whitespace
    split_keywords = [keyword.strip() for keyword in keywords.split(',')]
    # Add the split keywords to the set
    unique_keywords.update(split_keywords)

# Print each unique keyword
for keyword in unique_keywords:
    print(keyword)


machine learning
a.i.
ai/ml
large language model
GPT
artificial intelligence
nlp
image recognition
deep learning
data mining
data scientist
natural language processing
business intelligence
financial forecasting
big data
aiml
computer vision
machine translation
chatbot
llm
generative ai
algorithmic trading
data science


## QTR2:

In [30]:
df = pd.read_csv("/Users/rravindra0463/Desktop/ai_proj/venv/new_output/New_Results_QTR2.csv")
df.describe()

filename  \
count                                                  41   
unique                                                 41   
top     20230504_10-K_edgar_data_1213660_0001213900-23...   
freq                                                    1   

                                                   Item1A  \
count                                                  41   
unique                                                 41   
top     . RISK FACTORS Investing in our shares of Comm...   
freq                                                    1   

                                                 Sentence  \
count                                                  41   
unique                                                 41   
top     If we acquire other businesses, we may face di...   
freq                                                    1   

                                              SentencesR3  \
count                                                  41   
unique                                                 41   
top     Mergers and acquisitions of companies are inhe...   
freq                                                    1   

                                              SentencesR5  \
count                                                  41   
unique                                                 41   
top     If we fail to pay our indebtedness when due, i...   
freq                                                    1   

               KeywordMentioned  
count                        41  
unique                       29  
top     artificial intelligence  
freq                          6

In [33]:
df.head()

filename  \
0  20230504_10-K_edgar_data_1213660_0001213900-23...   
1  20230626_10-K_edgar_data_107140_0000107140-23-...   
2  20230524_10-K_edgar_data_733269_0000733269-23-...   
3  20230622_10-K_edgar_data_1576914_0001628280-23...   
4  20230413_10-K_edgar_data_1290658_0001683168-23...   

                                              Item1A  \
0  . RISK FACTORS Investing in our shares of Comm...   
1  . Risk Factors Introduction The risks describe...   
2  . Risk Factors An investment in our common sto...   
3  . RISK FACTORS Important factors that could ca...   
4  . Risk Factors Risks Related to Our Business A...   

                                            Sentence  \
0  If we acquire other businesses, we may face di...   
1  In addition, our ability to effectively compet...   
2  Without the timely introduction of new product...   
3  Inaccurate output from artificial intelligence...   
4  Aretas combines IoT sensor technology, cloud-b...   

                                         SentencesR3  \
0  Mergers and acquisitions of companies are inhe...   
1  In some cases, investments will take the form ...   
2  Advances in information technology are changin...   
3  Risk Factors. Risks related to Our Business We...   
4  There are restrictive provisions in the agreem...   

                                         SentencesR5  \
0  If we fail to pay our indebtedness when due, i...   
1  We must continue to invest in technology and o...   
2  If we fail to maintain our company culture, ou...   
3  The list below is not exhaustive, and investor...   
4  The agreement has an initial term of one year ...   

                                    KeywordMentioned  
0                              financial forecasting  
1                            artificial intelligence  
2          artificial intelligence, machine learning  
3  artificial intelligence, artificial intelligen...  
4                                   machine learning

In [35]:
# check how many files orginally present in the concerned folder,i.e QTR2 here
import os
folder_path = '/Users/rravindra0463/Desktop/ai_proj/venv/QTR2'

# list all files in the folder
files = [f for f in os.listdir(folder_path) if os.path.isfile(os.path.join(folder_path, f))]

# count the number of files
num_files = len(files)
print(f'Number of files in the folder QTR2: {num_files}')

# calculate metrics
unique_filenames = df['filename'].nunique()
percentage = (unique_filenames/ num_files) * 100

print(f'Number of relevant files extracted: {unique_filenames}')
print(f'Percentage of relevant files: {percentage:.2f}%')

Number of files in the folder QTR2: 7564
Number of relevant files extracted: 41
Percentage of relevant files: 0.54%


In [36]:
# Initialize an empty set to store unique keywords
unique_keywords = set()

# Iterate through each row in the 'KeywordMentioned' column
for keywords in df['KeywordMentioned'].dropna():
    # Split the keywords by comma and strip any extra whitespace
    split_keywords = [keyword.strip() for keyword in keywords.split(',')]
    # Add the split keywords to the set
    unique_keywords.update(split_keywords)

# Print each unique keyword
for keyword in unique_keywords:
    print(keyword)


machine learning
data scientist
artificial intelligence
generative ai
data mining
business intelligence
ai/ml
large language model
financial forecasting
big data
GPT
aiml
data science


## QTR3:

In [37]:
df = pd.read_csv("/Users/rravindra0463/Desktop/ai_proj/venv/new_output/New_Results_QTR3.csv")
df.describe()

filename  \
count                                                  51   
unique                                                 51   
top     20230811_10-K_edgar_data_943819_0000943819-23-...   
freq                                                    1   

                                                   Item1A  \
count                                                  51   
unique                                                 51   
top     RISK FACTORS Before deciding to purchase, hold...   
freq                                                    1   

                                                 Sentence  \
count                                                  51   
unique                                                 51   
top     Certain of our products and services include t...   
freq                                                    1   

                                              SentencesR3  \
count                                                  51   
unique                                                 51   
top     Failure to comply with these regulations could...   
freq                                                    1   

                                              SentencesR5  \
count                                                  51   
unique                                                 51   
top     Our business activities are subject to extensi...   
freq                                                    1   

               KeywordMentioned  
count                        51  
unique                       34  
top     artificial intelligence  
freq                         11

In [38]:
df.head()

filename  \
0  20230811_10-K_edgar_data_943819_0000943819-23-...   
1  20230721_10-K_edgar_data_866729_0000866729-23-...   
2  20230815_10-K_edgar_data_1049521_0001049521-23...   
3  20230824_10-K_edgar_data_1606757_0001606757-23...   
4  20230717_10-K_edgar_data_1048911_0000950170-23...   

                                              Item1A  \
0  RISK FACTORS Before deciding to purchase, hold...   
1  | Risk Factors Set forth below and elsewhere i...   
2  . RISK FACTORS: Risks Related to Business Oper...   
3  - Risk Factors The following important risk fa...   
4  . RIS K FACTORS In addition to the other infor...   

                                            Sentence  \
0  Certain of our products and services include t...   
1  There are substantial uncertainties associated...   
2  Customers conditioned to seek quarter-end disc...   
3  The operation of our business depends on effec...   
4  Adverse publicity (whether or not justified) r...   

                                         SentencesR3  \
0  Failure to comply with these regulations could...   
1  If SaaS providers are unable to provide these ...   
2  High quarterly book-ship ratios pressure our i...   
3  Quality or noncompliance failures could have a...   
4  FedEx is one of the most widely recognized, tr...   

                                         SentencesR5  \
0  Our business activities are subject to extensi...   
1  The Company outsources business processes to r...   
2  Any significant change in our customers purcha...   
3  However, in the event we fail to adhere to the...   
4  We depend on our strong reputation and the val...   

                                    KeywordMentioned  
0                            artificial intelligence  
1                            artificial intelligence  
2                              financial forecasting  
3  artificial intelligence, machine learning, mac...  
4  artificial intelligence, artificial intelligen...

In [39]:
# check how many files orginally present in the concerned folder,i.e QTR3 here
import os
folder_path = '/Users/rravindra0463/Desktop/ai_proj/venv/QTR3'

# list all files in the folder
files = [f for f in os.listdir(folder_path) if os.path.isfile(os.path.join(folder_path, f))]

# count the number of files
num_files = len(files)
print(f'Number of files in the folder QTR3: {num_files}')

# calculate metrics
unique_filenames = df['filename'].nunique()
percentage = (unique_filenames/ num_files) * 100

print(f'Number of relevant files extracted: {unique_filenames}')
print(f'Percentage of relevant files: {percentage:.2f}%')

Number of files in the folder QTR3: 6583
Number of relevant files extracted: 51
Percentage of relevant files: 0.77%


In [40]:
# Initialize an empty set to store unique keywords
unique_keywords = set()

# Iterate through each row in the 'KeywordMentioned' column
for keywords in df['KeywordMentioned'].dropna():
    # Split the keywords by comma and strip any extra whitespace
    split_keywords = [keyword.strip() for keyword in keywords.split(',')]
    # Add the split keywords to the set
    unique_keywords.update(split_keywords)

# Print each unique keyword
for keyword in unique_keywords:
    print(keyword)


machine learning
data scientist
artificial intelligence
generative ai
natural language processing
business intelligence
ai/ml
large language model
financial forecasting
big data
GPT
deep learning
data science


## QTR4:

In [41]:
df = pd.read_csv("/Users/rravindra0463/Desktop/ai_proj/venv/new_output/New_Results_QTR4.csv")
df.describe()

filename  \
count                                                  56   
unique                                                 56   
top     20231129_10-K_edgar_data_884614_0000884614-23-...   
freq                                                    1   

                                                   Item1A  \
count                                                  56   
unique                                                 56   
top     . RISK FACTORS There are many factors that may...   
freq                                                    1   

                                                 Sentence  \
count                                                  56   
unique                                                 55   
top     However, because the techniques used to obtain...   
freq                                                    2   

                                              SentencesR3  \
count                                                  56   
unique                                                 56   
top     In addition, the CPRA created a new state priv...   
freq                                                    1   

                                              SentencesR5  \
count                                                  56   
unique                                                 56   
top     The EU adopted the GDPR, which expanded EU dat...   
freq                                                    1   

               KeywordMentioned  
count                        56  
unique                       33  
top     artificial intelligence  
freq                         17

In [42]:
# check how many files orginally present in the concerned folder,i.e QTR4 here
import os
folder_path = '/Users/rravindra0463/Desktop/ai_proj/venv/QTR4'

# list all files in the folder
files = [f for f in os.listdir(folder_path) if os.path.isfile(os.path.join(folder_path, f))]

# count the number of files
num_files = len(files)
print(f'Number of files in the folder QTR4: {num_files}')

# calculate metrics
unique_filenames = df['filename'].nunique()
percentage = (unique_filenames/ num_files) * 100

print(f'Number of relevant files extracted: {unique_filenames}')
print(f'Percentage of relevant files: {percentage:.2f}%')

Number of files in the folder QTR4: 6565
Number of relevant files extracted: 56
Percentage of relevant files: 0.85%


In [43]:
# Initialize an empty set to store unique keywords
unique_keywords = set()

# Iterate through each row in the 'KeywordMentioned' column
for keywords in df['KeywordMentioned'].dropna():
    # Split the keywords by comma and strip any extra whitespace
    split_keywords = [keyword.strip() for keyword in keywords.split(',')]
    # Add the split keywords to the set
    unique_keywords.update(split_keywords)

# Print each unique keyword
for keyword in unique_keywords:
    print(keyword)


machine learning
chatbot
artificial intelligence
generative ai
algorithmic trading
business intelligence
a.i.
large language model
ai/ml
financial forecasting
GPT
big data
aiml
data science


### ------------------------------------------------------------------------ ###

<a href="https://drive.google.com/drive/folders/1-wpBG2rm4uG1lHidT3GddxIcE2awBAqG?usp=drive_link" target="_blank">Access all the csv files here!</a>
